e_methods.rb

```diff
@@ -0,0 +1,169 @@
+    elsif is_a?(Range)
+      result = []
+      my_each { |value| result.push(value) if yield(value) }
+    else
+      result = {}
+      my_each { |key, value| result[key] = value if yield(key, value) }
+    end
+    result
+  end
+
+  def my_all?(argum = nil)
+    if block_given? && !argum.nil?
+      puts 'warning: given block not used'
+      my_each { |element| return false unless element.is_a? argum }
+    elsif block_given?
+      my_each { |element| return false unless yield(element) }
+    elsif argum.nil?
+      my_each { |element| return false if element.nil? || element === false }
+    elsif !argum.nil? && (argum.is_a? Class)
+      my_each { |element| return false unless element.is_a? argum }
+    elsif !argum.nil? && argum.instance_of?(Regexp)
+      my_each { |element| return false unless argum.match(element) }
+    elsif !argum.nil?
+      my_each { |element| return false unless element == argum }
+    else
+      my_each { |element| return false unless element }
+    end
+    true
+  end
+
+  def my_any?(argum = nil)
+    if block_given? && !argum.nil? && argum.instance_of?(Regexp)
+      puts 'warning: given block not used'
+      my_each { |element| return true if argum.match(element) }
+    elsif !argum.nil? && (argum.is_a? Class)
+      puts 'warning: given block not used'
+      my_each { |element| return true if element.is_a? argum }
+    elsif block_given? && !argum.nil?
+      puts 'warning: given block not used'
+      my_each { |element| return true if element == argum }
+    elsif block_given?
+      my_each { |element| return true if yield(element) }
+    elsif argum.nil?
+      my_each { |element| return true if element.nil? || element == false }
+    elsif !argum.nil? && argum.instance_of?(Regexp)
+      my_each { |element| return true if argum.match(element) }
+    elsif argum
+      my_each { |element| return true if element == argum }
+    else
+      my_each { |element| return true if element }
+    end
+    false
+  end
+
+  def my_none?(argum = nil)
+    if block_given? && !argum.nil?
+      puts 'warning: given block not used'
+      my_each { |element| return true unless element.is_a? argum }
+    elsif is_a?(Range) && block_given?
+      to_a.my_each { |element| return true if yield(element) == false }
+    elsif block_given?
+      to_a.my_each { |element| return true unless yield(element) == true }
+    elsif argum.nil?
+      my_each { |element| return false if element == true }
+      my_each { |element| return true if element == false }
+    elsif !argum.nil? && (argum.is_a? Class)
+      my_each { |element| return true unless element.is_a? argum }
+    elsif !argum.nil? && argum.instance_of?(Regexp)
+      my_each { |element| return true unless argum.match(element) }
+    elsif argum
+      my_each { |element| return false if element == argum }
+      my_each { |element| return true if element != argum }
+    else
+      my_each { |element| return true unless element }
+    end
+    false
+  end
```

Comment:
## my_any?:

- [ ]  returns false if the block never returns true for all elements::range:

```ruby

      range = Range.new(5, 50)
      false_block = proc { |num| num > 51 }
      range.my_none?(&false_block) === range.none?(&false_block) => false

```